app/src/main/java/com/thailam/piggywallet/data/source/TransactionDataSource.java

```diff
@@ -19,4 +19,6 @@ void saveTransaction(Wallet wallet, Transaction transaction,
                          @NonNull TransactionCallback callback);
 
     void getInitialTransactions(int walletId, @NonNull GetTransactionCallback callback);
+
+    void saveWalletToSharedPref(Wallet wallet);
```

Comment:
save ko có kết quả thành công hay thất bại hả em

Reply:
Em dùng shared pref .apply trong docs nó ghi là không được notify error. Dùng commit() thì có boolean nhưng nó bảo nên dùng apply() hơn
![Uploading Screenshot from 2019-05-02 09-18-01.png…]()

Reply:
# https://developer.android.com/reference/android/content/SharedPreferences.Editor.html#apply()
As SharedPreferences instances are singletons within a process, it's safe to replace any instance of commit() with apply() if you were already ignoring the return value.
ở đây thì em nên sử dụng apply nếu em muốn ignoring the return value.
Còn nếu k ignore thì e sử dụng commit cũng được
Trường hợp nếu sử dụng apply thì nó sẽ mặc định là true.